protocols/v2/binary-sv2/no-serde-sv2/codec/src/datatypes/non_copy_data_types/seq_inner.rs

```diff
@@ -109,6 +109,12 @@ impl<'a, T: 'a> Seq064K<'a, T> {
             Err(Error::SeqExceedsMaxSize)
         }
     }
+
+    /// Return a reference to the inner vector of T with the same lifetime
+    /// guarantees as the struct.
+    pub fn inner_as_ref(&'a self) -> &'a Vec<T> {
```

Comment:
@Fi3 do you have a suggestion of renaming this method?

As far I can tell, the implementation of `inner_as_ref()` for `Seq0255` gets all inner T and returns a vector of all bytes of all T in sequence. So the iterator would be stepping by every individual byte instead of stepping by each `T`.

In this case, I want to get a `Vec<T>` and get each iteration of `T` and deserialize to a `Tx`: https://github.com/ccdle12/stratum/blob/2022.11.03-fix-cb-height-bug/protocols/v2/roles-logic-sv2/src/job_creator.rs#L220

Reply:
What about something like `map_inners` so that we can do `self.coinbase_outputs.map_inners(TxOut::deserialize)`

If you go with this solution please add it also for `Seq0255`, and (fi possible) for the respective types in the serde serializer/deserializer https://github.com/stratum-mining/stratum/tree/main/protocols/v2/binary-sv2/serde-sv2/src/primitives/sequences

Also this method should be tested here https://github.com/stratum-mining/stratum/blob/main/protocols/v2/binary-sv2/binary-sv2/src/lib.rs

Reply:
Also please note the  `inner_as_ref()` for `Seq0255` do not return a vectors of bytes but return a vector of slices. So I guess that if you want implement it exactly as we did for the `Seq0255` it will works for your use case.